### ISSUE TITLE
istio-agent non-daemon mode (generate only)

### ISSUE THREAD

**Issue by costinm** [istio/istio#28791]
We need to be able to run istio-agent - or a smaller standalone binary - to do the init and generate the 
certificate files, but not start envoy or keep running.

This is needed for apps like Prometheus, which can use the mTLS certificate directly.

It is also needed for gRPC 'proxyless'. In this mode, we would also generate the bootstrap file for
gRPC which is similar but smaller than Envoy bootstrap.

**Describe the feature request**

**Describe alternatives you've considered**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Additional context**

**Comment by markdroth** [comment]
As we discussed, for gRPC, I think we'll actually need daemon mode, because we'll need istio-agent to continue to update the cert files as the certs rotate.  So the main thing we need is for istio-agent to generate the gRPC bootstrap file, and it might make sense to split that into a separate issue.

In the future, we can collaborate on ways to eliminate the dependency on istio-agent.  This will probably involve having a CertificateProvider plugin that can speak the Istio CA API (https://github.com/istio/api/blob/master/security/v1alpha1/ca.proto) and adding the ability to set STS call creds for the communication with the xDS server.  At that point, we can move the code to generate the gRPC bootstrap file out of istio-agent and into a standalone bootstrap generator, which users can run on their nodes before starting their application.

**Comment by howardjohn** [comment]
Related: https://github.com/istio/istio/issues/28712

**Comment by costinm** [comment]
Agreed - I think we'll want both 'daemon' and 'exec/cron' mode ( similar with gcloud command ).

**Comment by costinm** [comment]
Updated requirement: 
- should be able to run as an XDS proxy ( performing istio-specific auth ), but without starting envoy
- instead of starting envoy, should start an app, using the reminder of the command line ( i.e. if an app
is started with "/usr/local/bin/myApp --foo --bar", we can use "istio-agent /usr/local/bin/myApp --foo --bar"

This will be used until gRPC ( or other 'proxyless' frameworks ) support all required Istio-specific auth
methods and CA protocol. The apps will still have no proxy in the data path.

**Comment by stevenctl** [comment]
I think the exact CLI needs some adjustments. We should have separate flags to say:
 1. Don't run envoy
 2.  Generate gRPC bootstrap (vs some other "proxyless" framework's bootstrap, or no bootstrap gen at all)
 3. Start some other binary using the other flags. 
 
 Something like `istio-agent --no-proxy --grpc-xds-bootstrap --exec /usr/local/bin/myApp --foo --bar`. 
 
 I'm not fully convinced we should have the agent manage the lifecycle of the user's app. Having the agent run in another container does make the integration feel less "proxyless" but I don't see any major advantages?

**Comment by ruigulala** [comment]
@costinm Can we clarify a bit regarding the usage of exec mode? As John mentioned in https://github.com/istio/istio/pull/33528, user can either run istio-agent with `--no-proxy --grpc-xds-bootstrap` and run the user app in a separate command or they can run the user command within istio-agent exec mode.

What's the advantage of the latter?

**Comment by costinm** [comment]
Added a comment to the PR - the goal is to have istio-agent start, create the workload cert and any other file required by the app, and exec the app after, so app has a valid environment. 

Otherwise the app will need to have complicated code to poll for the cert or XDS socket showing up, or fail - or relatively tricky shell scripts.

**Comment by costinm** [comment]
BTW - there is a use case for having exec even when envoy is used - for example a VM or app running in a single-container Docker, where we want to have the app start after everything is in a 'stable' state. 

I would just add code to check if the CLI has some unprocessed args ( after -- ), and use them to exec even in envoy mode.

**Comment by ruigulala** [comment]
Is there anything else need to be done for this Issue? Last time when I talked to @costinm , he mentioned that this is not needed.
https://github.com/istio/istio/pull/33528#issuecomment-940557910
